Encoding provider-related revisions

rpetrusha opened the pull request:
# Encoding provider-related revisions

## Summary

The documentation for `Encoding.GetEncoding` didn't clarify how encoding providers affected the method, nor did it indicate how it affected GetEncoding(0). 

In addition, the Encoding.Default property is implementation-specific.

There has also been some content loss from documentation migration from MSDN to GitHub. I've added some of it back.

Fixes #1837, #2531 

## Suggested Reviewers

@tarekgh @mklement0

Commit:
Improve Encoding.GetEncoding(0) documentation

Commit:
Address lack of clarity in EncodingProvider-related documentation.

Commit:
Corrected content dropout

tarekgh commented on xml/System.Text/Encoding.xml:
```diff
@@ -865,10 +865,18 @@
+
+### The Default property in the .NET Framework
+
+In the .NET Framework on the Windows desktop, the <xref:System.Text.Encoding.Default%2A> property always gets the system's active code page and creates a <xref:System.Text.Encoding> object that corresponds to it. This ANSI code page includes the ASCII character set along with additional characters, which vary by code page. Because all <xref:System.Text.Encoding.Default%2A> encodings lose data, co ...
```
>This ANSI code page includes  [](start = 213, length = 29)

Windows 10 in the near future will start support UTF8 as active codepage. so I would like to avoid ANSI in here

tarekgh replied:
maybe we can say "if the ACP is one of the ANSI codepages, then ....)

---
In reply to: [129648883](https://github.com/dotnet/docs/pull/2755#discussion_r129648883) [](ancestors = 129648883)

tarekgh replied:
Also we may need to review the rest of this doc for any place mentioning ANSI and try to reword it

---
In reply to: [129649192](https://github.com/dotnet/docs/pull/2755#discussion_r129649192) [](ancestors = 129649192,129648883)

tarekgh commented on xml/System.Text/Encoding.xml:
```diff
@@ -2647,18 +2655,26 @@
+
+- On .NET Core, if no no encoding provider has been registered or if no registered encoding provider handles a `codepage` value of 0, it returns the <xref:System.Text.UTF8> encoding.
+
+To determine the default code pages used on Windows systems, call the Windows [GetSystemDefaultLangID](http://msdn.microsoft.com/library/windows/desktop/dd318120.aspx) function from the full .NET Framework on the WIndows desktop. To determine the current ANSI code page on Windows systems, call the Windows [GetACP](http://msdn.microsoft.com/library/windows/desktop/dd318070.aspx) function from the  ...
```
>WIndows  [](start = 213, length = 8)

typo

tarekgh replied:
>call the Windows [GetSystemDefaultLangID](http://msdn.microsoft.com/library/windows/desktop/dd318120.aspx) [](start = 61, length = 106)

This is wrong. default lang id is different than ACP. You need to delete this part of the paragraph and keep only the part for GetACP

tarekgh replied:
>ANSI [](start = 255, length = 4)

please remove the word ANSI

tarekgh commented on xml/System.Text/Encoding.xml:
```diff
@@ -2795,14 +2808,21 @@
+
+- If one or more encoding providers have been registered, it returns the encoding of the last registered provider that has chosen to return a encoding when the <xref:System.Text.Encoding.GetEncoding%2A> method is passed a `codepage` argument of 0.     
+
+- On the .NET Framework, if no encoding provider has been registered, if the <xref:System.Text.Encoding.CodePagesEncodingProvider> is the registered encoding provider, or if no registered encoding provider handles a `codepage` value of 0, it returns the default ANSI code page in the operating system's regional and language settings.
```
>ANSI [](start = 262, length = 4)

change to "Active"

tarekgh commented on xml/System.Text/Encoding.xml:
```diff
@@ -3861,7 +3881,13 @@ The goal is to save this file, then open and decode it as a binary stream.
+
+Registering an encoding provider by using the <xref:System.Text.Encoding.RegisterProvider%2A> method also modifies the behavior of the [Encoding.GetEncoding(Int32)](<xref:System.Text.Encoding.GetEncoding(System.Int32)>) and [EncodingProvider.GetEncoding(Int32, EncoderFallback, DecoderFallback)](xref:System.Text.Encoding.GetEncoding(System.Int32,System.Text.EncoderFallback,System.Text.DecoderFallb ...
+
+- If the registered provider is the <xref:System.Text.CodePagesEncodingProvider>, the method returns the encoding that matches the system ANSI code page when running on the Windows operating system.
```
>ANSI [](start = 138, length = 4)

"Active"

Commit:
Addressed review comments

rpetrusha commented:
@tarekgh, I've addressed all of your comments and removed unnecessary or inaccurate mentions of ANSI. I'll send you links to the published topics on the staging site when the build has completed.

Commit:
Corrected bad link

Commit:
Corrected formatting

Commit:
Corrected bad xrefs

Commit:
Incorporated review comments
